From: Jakub Jirutka <jakub@jirutka.cz>
Date: Wed, 20 Oct 2021 00:18:47 +0200
Subject: [PATCH] Remove build-time check for polkit-gobject-1

polkit-gobject-1 is actually not needed for building elogind with PolKit
support; elogind binaries are not dynamically linked against polkit
libraries, there's only runtime dependency.

I've verified that building without polkit-dev installed won't change the
resulting binaries.

We need to remove polkit-dev dependency from elogind to avoid circular
dependency between polkit and elogind.

See: https://gitlab.alpinelinux.org/alpine/aports/-/issues/13095

--- a/meson.build
+++ b/meson.build
@@ -1148,21 +1148,21 @@
 install_polkit_pkla = false
 if want_polkit != 'false' and not skip_deps
         install_polkit = true
-
-        libpolkit = dependency('polkit-gobject-1',
-                               required : false)
-        if libpolkit.found() and libpolkit.version().version_compare('< 0.106')
-                message('Old polkit detected, will install pkla files')
-                install_polkit_pkla = true
-        endif
+# XXX-Patched: libpolkit is actually not needed for building with polkit support
+#        libpolkit = dependency('polkit-gobject-1',
+#                               required : false)
+#        if libpolkit.found() and libpolkit.version().version_compare('< 0.106')
+#                message('Old polkit detected, will install pkla files')
+#                install_polkit_pkla = true
+#        endif
 #if 1 /// Disable polkit completely if libpolkit is not there. See elogind issue #167
-        if not libpolkit.found()
-                if want_polkit != 'auto'
-                        error('Polkit requested but libpolkit was not found.')
-                endif
-                install_polkit = false
-                want_polkit    = false
-        endif
+#        if not libpolkit.found()
+#                if want_polkit != 'auto'
+#                        error('Polkit requested but libpolkit was not found.')
+#                endif
+#                install_polkit = false
+#                want_polkit    = false
+#        endif
 #endif // 1
 endif
 conf.set10('ENABLE_POLKIT', install_polkit)
